src/RMPI/mpidata.jl

```diff
@@ -217,13 +217,57 @@ end
 function LinearAlgebra.dot(x, md::MPIData)
     return MPI.Allreduce(localpart(x)⋅localpart(md), +, md.comm)
 end
-function LinearAlgebra.dot(x, lop, md::MPIData)
-    return MPI.Allreduce(dot(x, lop, localpart(md)), +, md.comm)
-end
-function LinearAlgebra.dot(md_left::MPIData, lop, md_right::MPIData)
-    # Arguments are swapped since lop * md_right is not an MPIData and MPIData should be on
-    # the right hand side.
-    return conj(dot(lop * md_right, md_left))
+
+function LinearAlgebra.dot(md_left, lop, md_right::MPIData)
```

Comment:
```suggestion
function LinearAlgebra.dot(md_left::MPIData, lop, md_right::MPIData)
# strategy: accumulate the dot product without allocating and intermediate vector
# spawns to other ranks are stored in buffers and communicated via MPI point-point communications
```

Reply:
Did you want to add an explanatory comment as I suggested? I cannot see it in the most recent version of the code. Maybe you can think of better wording than my suggestion :-)